test/parallel/test-policy-integrity.js

```diff
@@ -1,414 +1,397 @@
+    const h = crypto.createHash(algo);
+    h.update(body);
+    values.push(`${algo}-${h.digest('base64')}`);
+  }
+  {
+    const h = crypto.createHash(algo);
+    h.update(body.replace('\n', '\r\n'));
+    values.push(`${algo}-${h.digest('base64')}`);
+  }
+  return values;
 }
 
-const policyFilepath = path.join(tmpdir.path, 'policy');
-
-const packageFilepath = path.join(tmpdir.path, 'package.json');
-const packageURL = pathToFileURL(packageFilepath);
-const packageBody = '{"main": "dep.js"}';
-const policyToPackageRelativeURLString = `./${
-  path.relative(path.dirname(policyFilepath), packageFilepath)
-}`;
-
-const parentFilepath = path.join(tmpdir.path, 'parent.js');
-const parentURL = pathToFileURL(parentFilepath);
-const parentBody = 'require(\'./dep.js\')';
-
-const workerSpawningFilepath = path.join(tmpdir.path, 'worker_spawner.js');
-const workerSpawningURL = pathToFileURL(workerSpawningFilepath);
+const policyPath = './policy.json';
+const parentBody = {
+  commonjs: `
+    if (!process.env.DEP_FILE) {
+      console.error(
+        'missing required DEP_FILE env to determine dependency'
+      );
+      process.exit(33);
+    }
+    require(process.env.DEP_FILE)
+  `,
+  module: `
+    if (!process.env.DEP_FILE) {
+      console.error(
+        'missing required DEP_FILE env to determine dependency'
+      );
+      process.exit(33);
+    }
+    import(process.env.DEP_FILE)
+  `,
+};
 const workerSpawningBody = `
-const { Worker } = require('worker_threads');
-// make sure this is gone to ensure we don't do another fs read of it
-// will error out if we do
-require('fs').unlinkSync(${JSON.stringify(policyFilepath)});
-const w = new Worker(${JSON.stringify(parentFilepath)});
-w.on('exit', process.exit);
+  const path = require('path');
+  const { Worker } = require('worker_threads');
+  if (!process.env.PARENT_FILE) {
+    console.error(
+      'missing required PARENT_FILE env to determine worker entry point'
+    );
+    process.exit(33);
+  }
+  if (!process.env.DELETABLE_POLICY_FILE) {
+    console.error(
+      'missing required DELETABLE_POLICY_FILE env to check reloading'
+    );
+    process.exit(33);
+  }
+  const w = new Worker(path.resolve(process.env.PARENT_FILE));
+  w.on('exit', (status) => process.exit(status === 0 ? 0 : 1));
 `;
 
-const depFilepath = path.join(tmpdir.path, 'dep.js');
-const depURL = pathToFileURL(depFilepath);
-const depBody = '';
-const policyToDepRelativeURLString = `./${
-  path.relative(path.dirname(policyFilepath), depFilepath)
-}`;
+let nextTestId = 1;
+function newTestId() {
+  return nextTestId++;
+}
+tmpdir.refresh();
 
-fs.writeFileSync(parentFilepath, parentBody);
-fs.writeFileSync(depFilepath, depBody);
+let spawned = 0;
+const toSpawn = [];
+function queueSpawn(opts) {
+  toSpawn.push(opts);
+  drainQueue();
+}
 
-const tmpdirURL = pathToFileURL(tmpdir.path);
-if (!tmpdirURL.pathname.endsWith('/')) {
-  tmpdirURL.pathname += '/';
+function drainQueue() {
+  if (spawned < 100 !== true) {
```

Comment:
Is this really clearer than `if (spawned >= 100)`?

Reply:
don't really agree, but i'm fine changing it